Gemfile

```diff
@@ -1,6 +1,9 @@
 source "https://rubygems.org"
 
-gem "cocoapods", "~> 1.11.3"
+# You may use http://rbenv.org/ or https://rvm.io/ to install and use this version
+ruby ">= 2.6.10"
```

Comment:
I agree with @parasharrajat that we should get on Ruby 3 if possible. However, I suggest that we lock to a specific version of Ruby and cocoapods. The `Podfile.lock` may have a diff if you do `pod install` from a different version of cocoapods. I think that also broke pod caching in our CI, so we should lock that down to a specific version.

Also, why remove the `.ruby-version` file?

Reply:
@roryabraham @parasharrajat It was removed in the last version and now the ruby version is specified in `Gemfile` file, according to the [upgrade helper tool](https://react-native-community.github.io/upgrade-helper/?from=0.71.2&to=0.72.0).

> Also, why remove the .ruby-version file?

These two commits also give us some insight about what those decisions:
* https://github.com/facebook/react-native/commit/e57b6d11fa39aa96d92a7a6b8c91429974bef81c
* https://github.com/facebook/react-native/commit/0f56cee8e1fca9575e83f439274b83e01bdd98e2

> I agree with @parasharrajat that we should get on Ruby 3 if possible. However, I suggest that we lock to a specific version of Ruby and cocoapods.

Yes, I agree as well (as long Ruby 3 works, needs testing). In fact, a common problem that we have here is that some contributors do `pod install` normally and some use `bundle` to execute this task, which sometimes result in diffs in `Podfile.lock`.

Coming from this [issue](https://github.com/Expensify/App/issues/21318#issuecomment-1602973573), I agree we should change the instructions to tell contributors to use only `bundle` to avoid these problems.